File: Projeto Php/projeto_completo/src/Criacionais/FactoryMethod/BancoDoBrasil/BancoDoBrasilBoleto30Dias.php
<?php

namespace DesignPatterns\Criacionais\FactoryMethod\BancoDoBrasil;

use DesignPatterns\Criacionais\FactoryMethod\Boleto;

class BancoDoBrasilBoleto30Dias extends Boleto
{
    public function __construct(float $valor)
    {
        parent::__construct($valor);
        $this->juros = 0.05;
        $this->desconto = 0.02;
        $this->multa = 0.05;
    }
}

File: Projeto.net/Com.Padroes.De.Projeto/Src/Com.Padroes.De.Projeto.FactoryMethod/2-ProdutoConcreto/BancoDoBrasil/BancoDoBrasilBoleto30Dias.cs
using Com.Padroes.De.Projeto.FactoryMethod.ProdutoAbstrado;

namespace Com.Padroes.De.Projeto.FactoryMethod._2_ProdutoConcreto.BancoDoBrasil
{
    public class BancoDoBrasilBoleto30Dias : Boleto
    {
        public BancoDoBrasilBoleto30Dias(float valor) : base(valor)
        {
            Juros = 0.05f;
            Desconto = 0.02f;
            Multa = 0.5f;
        }
    }
}
